chia/util/db_wrapper.py

```diff
@@ -70,39 +70,52 @@ def _next_savepoint(self) -> str:
         self._savepoint_name += 1
         return name
 
+    @contextlib.asynccontextmanager
+    async def _savepoint(self, connection) -> AsyncIterator[None]:
+        name = self._next_savepoint()
+        await connection.execute(f"SAVEPOINT {name}")
+        try:
+            # TODO: maybe yield out something to make it possible to cancel the
+            #       savepoint other than raising an exception?
+            yield
+        except:  # noqa E722
+            await connection.execute(f"ROLLBACK TO {name}")
+            raise
+        finally:
+            # rollback to a savepoint doesn't cancel the transaction, it
+            # just rolls back the state. We need to cancel it regardless
+            await connection.execute(f"RELEASE {name}")
+
+    @contextlib.contextmanager
+    def _set_current_writer(self, writer: asyncio.Task) -> Iterator[None]:
```

Comment:
I think this context manager may be taking it a little bit too far (in terms of lines of code and ease-of-reading to utility ratio)

Reply:
Yeah, this is on the edge for me too.  But while reading the existing code I kept going back to the question of whether the multiple activities in a `finally:` block were ok or not.  Could have added another `try:`/`finally:` for them, or just drop this here.  Either way.  I guess at the present implementation here it wouldn't be "another `try`", it would just be the only `try`, so that doesn't really bug me.